### PR TITLE
28 setup python linter locally and in GitHub action

### LINKED ISSUE #28: Setup Python Linter Locally and in GitHub Action
Due Jul 22, 2022

### PR DESCRIPTION
**Pull request by EdwardHaoranLee**
Setup pylint and black formatter in pre-commit and in github action. 

### Motivation and context

To make sure all the code submitted is following standard.

### Describe your changes

- Added `./github/workflows` for automate checking when a PR is submitted and modifed. 
- Added `.pre-commit-config.yaml` for a pre-commit check.
- Added `.pylintrc` as the config file for pylint.
- Modified the whole code base regarding to the suggested changes from pylint and black.

### Tests performed

*Skipped.*

### Before submitting this PR, please make sure:
 - [x] Your code builds cleanly without any errors

### CODE REVIEW

**Review by Xiaorui-Huang — DISMISSED** [review]
My gosh, this linter is so damn strict

File: setup.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+pip install PyYAML
```
Comment (Xiaorui-Huang) [inline setup.sh:1]: When is this setup.sh file used? I don't see it in the yaml config...
I was thinking maybe we should instead we pipenv which uses the already existing Pipfile to manage dependencies

Comment (EdwardHaoranLee) [reply setup.sh:1]: I was using this setup.sh to make sure all the required packages (especially the pre-commit setting) are setup. Tbh I am not sure how to do that in a Pipfile but I have the instinct that that would be better. 

Would you be able to just update this in this branch? Once you update you could just approve and merge this PR.

Comment (Xiaorui-Huang) [reply setup.sh:1]: no problem

**Review by Xiaorui-Huang — APPROVED** [review]

**Pull request merged**